### PR TITLE
Remove Node.get_position_in_parent()

### PR DESCRIPTION
**Pull request by KoBeWi**
See #21802

### CODE REVIEW

**Comment by pycbouh** [comment]
Why remove this and not the other? It's shorter and perfectly aligns with the fact that the result will be used as an `index` argument in a function like `get_child`.

**Comment by KoBeWi** [comment]
> I agree, get_position_in_parent() is waaay too wordy...

It's more descriptive though. Maybe we could come up with a completely new name?

**Comment by aaronfranke** [comment]
`get_index_in_parent()` is what immediately occurred to me. `get_position_in_parent()` is a really poor choice of words because "position" means something else for Node2D and Node3D.

**Comment by KoBeWi** [comment]
How about `get_sibling_index`?

**Comment by aaronfranke** [comment]
I think `get_sibling_index` is not immediately clear that we're getting the index of this node among siblings, since it could also be read as "get the index of another sibling". `get_index_among_siblings`? I'm still leaning towards `get_index_in_parent`.

**Comment by pycbouh** [comment]
Well, if we must include a reference to the parent, then `get_index_in_parent` is our only option. Definitely not `sibling`. That implies that this index has something to do with current node's siblings.

In fact, I can't think of any adjective to put before the word `index` here, that wouldn't sound ambiguous or even more confusing than without any.

**Comment by hubbyist** [comment]
get_node_index() maybe?

**Comment by Xrayez** [comment]
> > I agree, get_position_in_parent() is waaay too wordy...
> 
> It's more descriptive though. Maybe we could come up with a completely new name?

That's what documentation is for, thanks goodness Godot has good accessibility for docs.

**Comment by aaronfranke** [comment]
Since `get_index()` seems good to many people, we should change this PR to be the opposite of what it currently is, keeping `get_index()` and removing `get_position_in_parent()`.

This way we can avoid having a duplicate methods for now, and we can maybe rename `get_index` later if it's desired.

**Comment by KoBeWi** [comment]
Heh, looks like `get_index()` was more commonly used in the code. The number of changed lines significantly dropped.

File: doc/classes/Node.xml [hunk 1]
```diff
@@ -331,7 +324,7 @@
 				Returns the time elapsed since the last physics-bound frame (see [method _physics_process]). This is always a constant value in physics processing unless the frames per second is changed via [member Engine.iterations_per_second].
 			</description>
 		</method>
-		<method name="get_position_in_parent" qualifiers="const">
+		<method name="get_index" qualifiers="const">
```
Comment (aaronfranke) [inline doc/classes/Node.xml:327]: `get_index` is no longer sorted properly in the documentation due to how this change was made.

File: scene/main/node.cpp [hunk 2]
```diff
@@ -2860,7 +2855,7 @@ void Node::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_pause_mode"), &Node::get_pause_mode);
 	ClassDB::bind_method(D_METHOD("can_process"), &Node::can_process);
 	ClassDB::bind_method(D_METHOD("print_stray_nodes"), &Node::_print_stray_nodes);
-	ClassDB::bind_method(D_METHOD("get_position_in_parent"), &Node::get_position_in_parent);
+	ClassDB::bind_method(D_METHOD("get_index"), &Node::get_index);
```
Comment (aaronfranke) [inline scene/main/node.cpp:2858]: This is a pointless diff, no need to remove line 2838 and re-add it here.

Same for the method declarations and definitions.

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: scene/main/node.h [hunk 3]
```diff
@@ -369,7 +368,7 @@ class Node : public Object {
 	void set_process_unhandled_key_input(bool p_enable);
 	bool is_processing_unhandled_key_input() const;
 
-	int get_position_in_parent() const;
+	int get_index() const;
```
Comment (aaronfranke) [inline scene/main/node.h:371]: Pointless removal on line 323 and addition here.

File: scene/main/node.cpp [hunk 4]
```diff
@@ -2042,7 +2038,7 @@ bool Node::get_scene_instance_load_placeholder() const {
 	return data.use_placeholder;
 }
 
-int Node::get_position_in_parent() const {
+int Node::get_index() const {
```
Comment (aaronfranke) [inline scene/main/node.cpp:2041]: Pointless removal on line 1917 and addition here.

**Review by aaronfranke — CHANGES_REQUESTED** [review]

**Review by aaronfranke — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by aaronfranke** [comment]
Just adding this for the record: This fixes #37556
